app/assets/javascripts/discourse/lib/uploads.js.es6

```diff
@@ -220,37 +228,33 @@ export function uploadIcon(staff) {
   return allowsAttachments(staff) ? "upload" : "far-image";
 }
 
-function uploadLocation(url) {
-  if (Discourse.CDN) {
-    url = Discourse.getURLWithCDN(url);
-    return /^\/\//.test(url) ? "http:" + url : url;
-  } else if (Discourse.S3BaseUrl) {
-    if (url.indexOf("secure-media-uploads") === -1) {
-      return "https:" + url;
-    }
-    return window.location.protocol + url;
-  } else {
-    var protocol = window.location.protocol + "//",
-      hostname = window.location.hostname,
-      port = window.location.port ? ":" + window.location.port : "";
-    return protocol + hostname + port + url;
-  }
+function imageMarkdown(upload) {
+  return `![${markdownNameFromFileName(upload.original_filename)}|${
+    upload.thumbnail_width
+  }x${upload.thumbnail_height}](${upload.short_url || upload.url})`;
+}
+
+function playableMediaMarkdown(upload, type) {
+  return `![${markdownNameFromFileName(upload.original_filename)}|${type}](${
+    upload.short_url
+  })`;
+}
+
+function attachmentMarkdown(upload) {
+  return `[${upload.original_filename}|attachment](${
+    upload.short_url
+  }) (${I18n.toHumanSize(upload.filesize)})`;
 }
 
 export function getUploadMarkdown(upload) {
   if (isAnImage(upload.original_filename)) {
```

Comment:
Maybe we should rename this method to `isImage` so that it's the same as `isAudio` and `isVideo`?